wagtail_localize/views/update_translations.py

```diff
@@ -0,0 +1,109 @@
+        label=__("Publish immediately"),
+        help_text=__("This will apply the updates and publish immediately, before any new translations happen."),
+        required=False,
+    )
+
+
+class UpdateTranslationsView(SingleObjectMixin, TemplateView):
+    template_name = "wagtail_localize/admin/update_translations.html"
+    title = __("Update existing translations")
+
+    def get_object(self):
+        return get_object_or_404(TranslationSource, id=self.kwargs['translation_source_id'])
+
+    def get_title(self):
+        return self.title
+
+    def get_subtitle(self):
+        return self.object.object_repr
+
+    def get_form(self):
+        if self.request.method == 'POST':
+            return UpdateTranslationsForm(self.request.POST)
+        else:
+            return UpdateTranslationsForm()
+
+    def get_success_message(self):
+        return _("Successfully updated translations for '{object}'").format(object=self.object.object_repr)
+
+    def get_success_url(self):
+        return get_valid_next_url_from_request(self.request)
+
+    def get_default_success_url(self):
+        instance = self.object.get_source_instance()
+
+        if isinstance(instance, Page):
+            return reverse("wagtailadmin_explore", args=[instance.get_parent().id])
+        else:
+            return reverse("wagtailsnippets:edit", args=[instance._meta.app_label, instance._meta.model_name, quote(instance.pk)])
+
+    def get_edit_url(self, instance):
+        if isinstance(instance, Page):
+            return reverse("wagtailadmin_pages:edit", args=[instance.id])
+        else:
+            return reverse("wagtailsnippets:edit", args=[instance._meta.app_label, instance._meta.model_name, quote(instance.pk)])
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context.update({
+            "translations": [
+                {
+                    "title": str(translation.get_target_instance()),
+                    "locale": translation.target_locale,
+                    "edit_url": self.get_edit_url(translation.get_target_instance()),
+                }
+                for translation in self.object.translations.select_related("target_locale")
+            ],
+            "form": self.get_form(),
+            "next_url": self.get_success_url(),
+            "back_url": self.get_success_url() or self.get_default_success_url(),
+        })
+        return context
+
+    def post(self, request, **kwargs):
+        form = self.get_form()
+
+        if form.is_valid():
+            with transaction.atomic():
+                self.object.update_from_db()
+
+                if form.cleaned_data['publish_translations']:
+                    for translation in self.object.translations.select_related("target_locale"):
+                        translation.save_target(user=request.user, publish=True)
+
+                # TODO: Button that links to page in translations report when we have it
```

Comment:
Remind on the TODO for this button

Reply:
Will do this once I've added the report!